source/api.py

```diff
@@ -164,6 +164,11 @@ def setFocusObject(obj: NVDAObjects.NVDAObject) -> bool:  # noqa: C901
 		obj.treeInterceptor=treeInterceptorObject
 	else:
 		obj.treeInterceptor=None
+	if obj.treeInterceptor:

```

Comment:
Isn't this block executed on every focus change? I guess that's problematic.

Reply:
Agreed, this needs needs to be changed. It's probably worth caching the previous browse mode value